lib/spack/spack/hooks/resolve_shared_libraries.py

```diff
@@ -0,0 +1,238 @@
+# Copyright 2013-2024 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import fnmatch
+import io
+import os
+import re
+from typing import Dict, List, Union
+
+import llnl.util.tty as tty
+from llnl.util.filesystem import BaseDirectoryVisitor, visit_directory_tree
+from llnl.util.lang import stable_partition
+
+import spack.config
+import spack.error
+import spack.util.elf as elf
+
```

Comment:
```suggestion
from enum import StrEnum

class Policy(StrEnum):
    IGNORE = auto()
    WARN = auto()
    ERROR = auto()
```

Reply:
Requires 3.11+. I can live without enums since spack.config.get produces strings anyhow

Reply:
Oh shoot, I missed that. I saw the note on it "changed in 3.11" but early in the docs it notes it was added in 3.11.

It is probably fine, this is all self contained in the module here so the special-ness of the strings is probably not going to be a huge issue.

Reply:
Thanks for pointing it out, I didn't know it existed